src/coordination/coordinator_state_manager.cpp

```diff
@@ -31,37 +33,59 @@ CoordinatorStateManager::CoordinatorStateManager(CoordinatorInstanceInitConfig c
 
   cluster_config_ = cs_new<cluster_config>();
   cluster_config_->get_servers().push_back(my_srv_config_);
+  utils::EnsureDirOrDie(config.durability_dir);
+  kv_store_ = std::make_unique<kvstore::KVStore>(config.durability_dir);
 }
 
 auto CoordinatorStateManager::load_config() -> ptr<cluster_config> {
-  // Just return in-memory data in this example.
-  // May require reading from disk here, if it has been written to disk.
-  spdlog::trace("Loading cluster config");
+  MG_ASSERT(kv_store_, "Durability folder should be created.");
+  spdlog::trace("Loading cluster config from RocksDb");
+  auto servers = kv_store_->Get("servers");
+  if (!servers.has_value()) {
+    spdlog::trace("Didn't find anything stored on disk for cluster config.");
+    return cluster_config_;
+  }
+  spdlog::trace("Loading cluster config from disk.");
+  auto const json = nlohmann::json::parse(servers.value());
+  auto const real_servers = json.get<std::vector<std::tuple<int, std::string, std::string>>>();
+  cluster_config_->get_servers().clear();
+  for (auto const &real_server : real_servers) {
+    auto const &[coord_id, endpoint, aux] = real_server;
+    spdlog::trace("Recreating cluster config with id: {}, endpoint: {} and aux data: {} from disk.", coord_id, endpoint,
+                  aux);
+    auto one_server_config = cs_new<srv_config>(coord_id, 0, endpoint, aux, false);
```

Comment:
You can move endpoint and aux. Aux is a big string (json) so that should improve perf

Reply:
Yes, makes sense